src/daft-sql/src/modules/utf8.rs

```diff
@@ -1,313 +1,517 @@
+                $doc.to_string()
+            }
+
+            fn arg_names(&self) -> &'static [&'static str] {
+                &[$arg_name]
+            }
+        }
+    };
+    ($name:ident, $sql_name:expr, $func:expr, $doc:expr, $arg_name_1:expr, $arg_name_2:expr) => {
+        pub struct $name;
+        impl SQLFunction for $name {
+            fn to_expr(
+                &self,
+                inputs: &[sqlparser::ast::FunctionArg],
+                planner: &crate::planner::SQLPlanner,
+            ) -> SQLPlannerResult<ExprRef> {
+                utf8_binary($func, $sql_name, $arg_name_1, $arg_name_2, inputs, planner)
+            }
+
+            fn docstrings(&self, _alias: &str) -> String {
+                $doc.to_string()
+            }
+
+            fn arg_names(&self) -> &'static [&'static str] {
+                &[$arg_name_1, $arg_name_2]
+            }
+        }
+    };
+    ($name:ident, $sql_name:expr, $func:expr, $doc:expr, $arg_name_1:expr, $arg_name_2:expr, $arg_name_3:expr) => {
+        pub struct $name;
+        impl SQLFunction for $name {
+            fn to_expr(
+                &self,
+                inputs: &[sqlparser::ast::FunctionArg],
+                planner: &crate::planner::SQLPlanner,
+            ) -> SQLPlannerResult<ExprRef> {
+                utf8_ternary(
+                    $func,
+                    $sql_name,
+                    $arg_name_1,
+                    $arg_name_2,
+                    $arg_name_3,
+                    inputs,
+                    planner,
+                )
+            }
+
+            fn docstrings(&self, _alias: &str) -> String {
+                $doc.to_string()
+            }
+
+            fn arg_names(&self) -> &'static [&'static str] {
+                &[$arg_name_1, $arg_name_2, $arg_name_3]
+            }
+        }
+    };
+}
+
 pub struct SQLModuleUtf8;
 
 impl SQLModule for SQLModuleUtf8 {
     fn register(parent: &mut crate::functions::SQLFunctions) {
-        use Utf8Expr::{
-            Capitalize, Contains, EndsWith, Extract, ExtractAll, Find, Left, Length, LengthBytes,
-            Lower, Lpad, Lstrip, Match, Repeat, Replace, Reverse, Right, Rpad, Rstrip, Split,
-            StartsWith, ToDate, ToDatetime, Upper,
-        };
-        parent.add_fn("ends_with", EndsWith);
-        parent.add_fn("starts_with", StartsWith);
-        parent.add_fn("contains", Contains);
-        parent.add_fn("split", Split(false));
+        parent.add_fn("ends_with", SQLUtf8EndsWith);
+        parent.add_fn("starts_with", SQLUtf8StartsWith);
+        parent.add_fn("contains", SQLUtf8Contains);
+        parent.add_fn("split", SQLUtf8Split);
         // TODO add split variants
         // parent.add("split", f(Split(false)));
-        parent.add_fn("regexp_match", Match);
-        parent.add_fn("regexp_extract", Extract(0));
-        parent.add_fn("regexp_extract_all", ExtractAll(0));
-        parent.add_fn("regexp_replace", Replace(true));
```

Comment:
@universalmind303 I think this should be a bug. The `regexp_replace` set the `regexp` to True, while the [implementation](https://github.com/Eventual-Inc/Daft/pull/3101/files#diff-06e4a592259b845ed2b29b25e79451299bf1bd8947755f6752bd895610bfe88fL209) is not.